presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java

```diff
@@ -51,264 +43,104 @@ private RegexpFunctions()
 
     @ScalarOperator(OperatorType.CAST)
     @SqlType(RegexpType.NAME)
-    public static Regex castToRegexp(@SqlType(StandardTypes.VARCHAR) Slice pattern)
+    public static RegexpGenericPattern castToRegexp(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice pattern)
     {
-        Regex regex;
+        String regexLibraryName = session.getSystemProperty(REGEX_LIBRARY, String.class);
+        RegexLibrary regexLibrary;
+        if (regexLibraryName.toUpperCase().equals("RE2J")) {
+            regexLibrary = RE2J;
+        }
+        else if (regexLibraryName.toUpperCase().equals("JONI")) {
+            regexLibrary = JONI;
+        }
+        else {
+            throw new PrestoException(INVALID_SESSION_PROPERTY, "The setting for '" + REGEX_LIBRARY + "' is not valid. It should be either RE2J or JONI.");
+        }
+
+        int dfaStatesLimit = session.getSystemProperty(RE2J_DFA_STATES_LIMIT, Integer.class);
```

Comment:
We forked it and implemented DFA support in our fork as well as other optmizations.  That's the artifact released as re2j-td.  (see https://github.com/Teradata/re2j)

It's not merged back to google/re2j because it operates on slices, which breaks compatibility with the current api.  (https://github.com/google/re2j/issues/17)